pkg/infra/server/baseServer.go

```diff
@@ -12,25 +13,11 @@ type BaseServer struct {
 	Agents    []baseAgent.Agent
 }
 
-type AgentGenerator func() baseAgent.Agent
-
-type AgentGeneratorCountPair struct {
-	generator AgentGenerator
-	count     int
-}
-
-func (bs *BaseServer) Init() {
+func (bs *BaseServer) Init(m []AgentGeneratorCountPair) {
 	fmt.Println("Server Init")
-	bs.NumAgents = 5
-	bs.NumTurns = 4
-	bs.Agents = make([]baseAgent.Agent, bs.NumAgents)
-	for i := 0; i < bs.NumAgents; i++ {
-		//converts the iteration to string
-		name := strconv.Itoa(i)
-		//creates new Agent
-		bs.Agents[i] = baseAgent.NewAgent(name)
-
-	}
+	initialisedAgents := bs.initAgents(m)
+	bs.Agents = initialisedAgents
+	bs.NumAgents = len(initialisedAgents)
```

Comment:
pretty good, but this shouldn't be part of the interface, nor exported. Only the package should handle this part, as we don't want users doing something weird